### PR TITLE
KAFKA-16671: enable test for ensureInternalEndpointIsSecured

### PR DESCRIPTION
**Pull request by FrankYang0529**
Related JIRA: https://issues.apache.org/jira/browse/KAFKA-16671

Enable the test to see error message. If there is any error, I will close this PR and investigate how to fix. If not, I will change the title to KAFKA-16671.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@FrankYang0529 please fix the build error

File: connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java [hunk 1]
```diff
@@ -115,11 +112,11 @@ public void ensureInternalEndpointIsSecured() throws Throwable {
         // Try again, but with an invalid signature
         log.info(
             "Making a POST request to the {} endpoint with no connector started and an invalid signature header; "
-                + "expecting 403 error response",
+                + "expecting 503 error response",
             connectorTasksEndpoint
         );
         assertEquals(
-            FORBIDDEN.getStatusCode(),
+            SERVICE_UNAVAILABLE.getStatusCode(),
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: Could you share the details with me?

Comment (FrankYang0529) [reply connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: Here, if the connector is not started, there is no `sessionKey`.

https://github.com/apache/kafka/blob/05df10449eb9c95fe6d6055b302c84686be8058d/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L2748-L2752

Comment (FrankYang0529) [reply connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: This was added in https://github.com/apache/kafka/pull/11783.

Comment (FrankYang0529) [reply connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: CI can't pass. Not sure whether there is some connector forgot to close. The error is:

```
java.lang.AssertionError: expected:<503> but was:<403>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.kafka.connect.integration.SessionedProtocolIntegrationTest.ensureInternalEndpointIsSecured(SessionedProtocolIntegrationTest.java:118)
```

Comment (chia7712) [reply connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: `503` is a temporary error, because the worker is still in starting. Could you please add a wait to check the 503 first ?

Comment (FrankYang0529) [reply connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: Added it and `ensureInternalEndpointIsSecured` can pass in CI environment. Thanks.

File: connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java [hunk 2]
```diff
@@ -108,8 +110,8 @@ public void ensureInternalEndpointIsSecured() throws Throwable {
             connectorTasksEndpoint
         );
         assertEquals(
-            BAD_REQUEST.getStatusCode(),
-            connect.requestPost(connectorTasksEndpoint, "[]", emptyHeaders).getStatus()
+                BAD_REQUEST.getStatusCode(),
```
Comment (chia7712) [inline connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: please revert those unrelated changes

Comment (FrankYang0529) [reply connect/runtime/src/test/java/org/apache/kafka/connect/integration/SessionedProtocolIntegrationTest.java]: Revert it. Thanks.
